homeassistant/components/network/models.py

```diff
@@ -24,7 +24,7 @@ class Adapter(TypedDict):
     """Configured network adapters."""
 
     name: str
-    index: int
+    index: int | None
```

Comment:
What is that change for? I do not find any other change where index is used and might be None.

Reply:
This is because `IP.Adapter.index` is an optional int.

https://github.com/home-assistant/core/pull/93377/files#diff-5e2ced21c93226c13b2938a7de1c13424c8600430d945907bf64cb6cf9579520R111

There were some changes where type hints were added.